### PR TITLE
UART LED control guide

### PR DESCRIPTION
**Pull request by gabbapeople**
Update `Case studies` with the `UART LED control example`

### CODE REVIEW

**Comment by gabbapeople** [comment]
- Proper screenshot heights
- Proper *.png sizes
- Full URL paths for libs

File: docs/guide/uart-led-control/README.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+  avoid confusion, label them differently. For example, `pot A` and `pot B`.
+- Set the `PORT` pins of the `pot` nodes. As we connected them to `A0` and `A1`,
+  we set the`PORT` pin values to `A0` and `A1`.
+- A `pot` node outputs the signal values in the range from `0` to `1`. A
+  potentiometer is an analog electronic component. Even if you don’t touch the
+  potentiometer knob, the ‘pot’ output value can continuously change for example
+  from `0.855` to `0.856` and back. To get rid of this minor changes lets round
+  the potentiometers outputs to two decimal places after the dot.
+- Add two `round-fraction` nodes onto the patch and link them with `pot` nodes.
+  To set two decimal places after dot place the `2` value to the `DIG` pin.
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: :scroll:: …after the dot, place…

File: docs/guide/uart-led-control/README.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+from potentiometers separated by a comma.
+
+Here is an example of the string: `0.25,0.95`.
+
+![Pots patch step 2](./pots.step2.patch.png)
+
+To combine all data values to a single string, use the
+[`join`](https://xod.io/libs/xod/core/join/) node.
+
+- Place ‘join’ onto the patch.
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: wrong quotes

File: docs/guide/uart-led-control/README.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+  [`soft-uart`](https://xod.io/libs/xod/uart/soft-uart/) node from the
+  `xod/uart` library onto the patch. Set the `RX` and `TX` pin values. We put
+  the `D8` value to the `RX` pin and the `D9` value to the `TX`.
+- The `BAUD` pin sets the data rate in bits per second (baud) for serial
+  communication. These baud values are standardized. For the software serial
+  interface, the maximum baud value is `115200`. Note that the higher the baud
+  value, the greater the chance that a data may get lost or broken during the
+  transmission. We set the `BAUD` value to `19200`.
+- Add the [`print`](https://xod.io/libs/xod/uart/print/) node from the
+  `xod/uart` library into the patch. This node transforms a String value from
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: Either: “a string value” or “a `String` value”

File: docs/guide/uart-led-control/README.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+- Link the `DATA` pin with the `join` node.
+- Link the`UART` pin of the `print` node with the `soft-uart` node.
+
+Here is an example of the line to be sent: `0.25,0.95\r\n`.
+
+### Limit frequency
+
+The last step for the `pots` patch is to limit the sending frequency. We will
+send a new line each time any of the potentiometer values changes, also we are
+going to use a `throttle` node to slow down the pulse frequency.
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: Suggest:

(1) ”each time any” → “only when any”.
(2) start a new paragraph at “also we are going”

File: docs/guide/uart-led-control/README.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+During the turn, the potentiometer generates lots of new values and a pulse rate
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: :scroll: a potentiometer

:scroll: the pulse rate

File: docs/guide/uart-led-control/README.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+- Put the [`any`](https://xod.io/libs/xod/core/any/) node onto the patch and
+  link its inputs with the `pulse-on-change` nodes. Now, the `any` and
+  `pulse-on change` nodes will generate a pulse if the potentiometer value is
+  changed.
+- Add the `throttle` node onto the patch and place it between `any` and `print`
+  nodes. `throttle` passes a pulse through no more than once every `T` seconds.
+  To slow down the frequency we place the `0.05` seconds value to the `T` pin.
+
+The patch for the first Arduino board is done. Let’s make the next one.
+
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: Suggest a reader: “Upload, use a terminal emulator to observe the result, wow!”

File: docs/guide/uart-led-control/README.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+![LEDS patch step 1](./leds.step1.patch.png)
+
+- Open the `leds` patch.
+- To set up the software serial interface put the
+  [`soft-uart`](https://xod.io/libs/xod/uart/soft-uart/) node from the
+  `xod/uart` library onto the patch. Set the `RX` and `TX` pin values. We put
+  the `D8` value to the `RX` pin and the `D9` value to the `TX`. -Arduino boards
+  must communicate at the same speed. The `BAUD` pin value is the same as on the
+  previous patch. Put the `19200` value to the `BAUD` pin. -Add the `read-line`
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: Bad formatting

File: docs/guide/uart-led-control/README.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+  onto the patch. This node listens to the UART interface for the stream of
+  bytes. If it is available, the `read-line` node searches the `\n` byte literal
+  in it and dumps all the characters before the ‘\n’ to a single line. It also
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: Bad quotes

File: docs/guide/uart-led-control/README.md [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+The program for the Arduino board with the LEDs starts with receiving data from
+UART.
+
+![LEDS patch step 1](./leds.step1.patch.png)
+
+- Open the `leds` patch.
+- To set up the software serial interface put the
+  [`soft-uart`](https://xod.io/libs/xod/uart/soft-uart/) node from the
+  `xod/uart` library onto the patch. Set the `RX` and `TX` pin values. We put
+  the `D8` value to the `RX` pin and the `D9` value to the `TX`. -Arduino boards
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: Bad formatting

File: docs/guide/uart-led-control/README.md [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+  must communicate at the same speed. The `BAUD` pin value is the same as on the
+  previous patch. Put the `19200` value to the `BAUD` pin. -Add the `read-line`
+  onto the patch. This node listens to the UART interface for the stream of
+  bytes. If it is available, the `read-line` node searches the `\n` byte literal
+  in it and dumps all the characters before the ‘\n’ to a single line. It also
+  cuts the `\r` byte literal from the output. The `READ` pin triggers a new read
+  of a line. -Link the `UART` pin of the `read-line` node with the `soft-uart`
```
Comment (nkrkv) [inline docs/guide/uart-led-control/README.md]: ditto

File: docs/guide/uart-led-control/README.md (hunk 7 above)
Comment (nkrkv) [reply docs/guide/uart-led-control/README.md]: Missing new line before `-`. Please, review how this list renders in GitHub [preview](https://github.com/xodio/xod-docs/blob/fd79e341437d47908a652d17ba51051ce88aea5f/docs/guide/uart-led-control/README.md)

File: docs/guide/uart-led-control/README.md [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+- Add the `throttle` node onto the patch and place it between `any` and `print`
+  nodes. `throttle` passes a pulse through no more than once every `T` seconds.
+  To slow down the frequency we place the `0.05` seconds value to the `T` pin.
+
+The patch for the first Arduino board is done. Upload, use a terminal emulator
+to observe the result. Let’s make the next patch.
+
+### Receive data
+
+The program for the Arduino board with the LEDs starts with receiving data from
```
Comment (brusherru) [inline docs/guide/uart-led-control/README.md]: `with LEDs`?

File: docs/guide/uart-led-control/README.md [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+UART. Create a new patch for the LEDs connected board and name it `leds`.
+
+![LEDS patch step 1](./leds.step1.patch.png)
+
+- To set up the software serial interface put the
+  [`soft-uart`](https://xod.io/libs/xod/uart/soft-uart/) node from the
+  `xod/uart` library onto the patch. Set the `RX` and `TX` pin values. We put
```
Comment (brusherru) [inline docs/guide/uart-led-control/README.md]: "Set ~~the~~ `RX` and `TX` pin values"?

File: docs/guide/uart-led-control/README.md [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+
+</div>
+
+![Receiver board circuit](./arduino-leds.fz.png)
+
+### Link Arduino boards
+
+The third step is to create a UART connection between the Arduino boards. All
+Arduino boards have at least one UART interface also known as _hardware_ serial
+or just “serial.”
```
[the remark this answers is not in the document]
Comment (nkrkv) [reply docs/guide/uart-led-control/README.md]: Yes, English is so “awkward.” It is correct.

**Pull request merged**